core/modules/widgets/widget.js

```diff
@@ -413,16 +413,23 @@ Widget.prototype.getAttribute = function(name,defaultText) {
 };
 
 /*
-Assign the computed attributes of the widget to a domNode
+Assign the common attributes of the widget to a domNode
 options include:
-excludeEventAttributes: ignores attributes whose name begins with "on"
+sourcePrefix: prefix of attributes that are to be directly assigned (defaults to the emtpy string meaning all attributes)
+destPrefix: prefix to be applied to attribute names that are to be directly assigned (defaults to the emtpy string which means no prefix is added)
+changedAttributes: hashmap by attribute name of attributes to process (if missing, process all attributes)
+excludeEventAttributes: ignores attributes whose name would begin with "on"
 */
 Widget.prototype.assignAttributes = function(domNode,options) {
-	options = options || {};
 	var self = this;
+	options = options || {};
+	var changedAttributes = options.changedAttributes || this.attributes;
+	var sourcePrefix = options.sourcePrefix || "";
+	var destPrefix = options.destPrefix || "";
+	var EVENT_ATTRIBUTE_PREFIX = "on";
 	var assignAttribute = function(name,value) {
```

Comment:
Yikes. I am surprised we never caught this earlier. This function is never called and we are assign attributes directly to the DOM node with no sanitation of event attributes!

Reply:
We should add a test that explicitly checks to make sure that event attributes are not supported.

Reply:
Ouch! Thanks @saqimtiaz